app/views/middleware_server/_deploy.html.haml

```diff
@@ -0,0 +1,40 @@
+- @angular_form = true
+
+%form.form-horizontal#form_deploy{"name"          => "angularForm",
+                                  "ng-controller" => "middlewareAddDeploymentController"}
+  .modal.fade#modal_d_div{"tabindex"        => "-1",
+                          "role"            => "dialog",
+                          "aria-labelledby" => "mw_modal_label",
+                          "aria-describedby" => "modal",
+                          "aria-hidden"     => "true",
+                          "data-keyboard"   => "false",
+                          "data-backdrop"   => "static",
+                          :style            => "display: none"}
+    .modal-dialog.modal-lg
+      .modal-content#deploy_div
+        .modal-header
+          %button.close{"data-dismiss" => "modal"}
+            %span{"aria-hidden" => "true"}
+              &times;
+            %span.sr-only
+              Close
+          %h4.modal-title#mw_modal_label
+            = _("Add Middleware Deployment")
+        .modal-body
+          %input#server_id{:type => "hidden", :value => params[:id]}
+          Select the file to deploy:
+          = file_field("upload", "file",
+                       :class             => "lifestyle",
+                       "data-buttonName"  => "btn-default",
+                       "ng-model"         => "filePath",
+                       "require-file"     => "handle")
+          :javascript
+            $(":file").filestyle({icon: false, placeholder: "No file chosen"});
```

Comment:
Multiple other places are using the file selector like this, so I didn't changed it as I didn't knew how/if it would work. Might have some trick as it is JS in a HAML file.